checker/src/main/java/org/checkerframework/checker/mustcall/MustCallAnnotatedTypeFactory.java

```diff
@@ -413,7 +413,9 @@ public Void visitIdentifier(IdentifierTree tree, AnnotatedTypeMirror type) {
       Element elt = TreeUtils.elementFromUse(tree);
       if (elt.getKind() == ElementKind.PARAMETER
           && (noLightweightOwnership || getDeclAnnotation(elt, Owning.class) == null)) {
-        type.replaceAnnotation(BOTTOM);
+        if (!type.hasAnnotation(POLY)) {
```

Comment:
Could we add a comment explaining this new logic?

Reply:
I've added a comment explaining it.

For more context, though, this change makes this code (which treats non-owning parameters inside a method body as bottom) analogous to the similar code at call sites (which treats all non-owning parameter types as top). That code already had an exceptional for polymorphic types, so this change just makes the treatment inside method bodies the same.

Reply:
(The similar code I mentioned is implemented by `MustCallAnnotatedTypeFactory#changeNonOwningParameterTypesToTop`, if you'd like to look at it.)

Reply:
Makes sense thanks!